src/generator.cs

```diff
@@ -1248,13 +1266,14 @@ public class MemberInformation
 	public readonly MemberInfo mi;
 	public readonly Type type;
 	public readonly Type category_extension_type;
-	public readonly bool is_abstract, is_protected, is_internal, is_unified_internal, is_override, is_new, is_sealed, is_static, is_thread_static, is_autorelease, is_wrapper, is_forced;
+	public readonly bool is_abstract, is_protected, is_internal, is_unified_internal, is_override, is_new, is_sealed, is_static, is_thread_static, is_autorelease, is_wrapper, is_forced, is_bindAs, is_bindAs_nullable;
 	public readonly bool is_type_sealed;
 	public readonly Generator.ThreadCheck threadCheck;
 	public bool is_unsafe, is_virtual_method, is_export, is_category_extension, is_variadic, is_interface_impl, is_extension_method, is_appearance, is_model, is_ctor;
 	public bool is_return_release;
 	public bool protocolize;
 	public string selector, wrap_method, is_forced_owns;
+	public readonly Type bindAs;
```

Comment:
Actually I think you need to be more specific than this, because here you only support 1 `[BindAs]` attribute per method, but what if you'd want to do this:

    [Export ("multiMethod:second:")]
    NSNumber MultiMethod ([BindAs (typeof (int?)) NSNumber arg1, [BindAs (typeof (long?)) NSNumber arg2);